Penteract/Assets/Scripts/HUDManager.cpp

```diff
@@ -0,0 +1,1210 @@
+			rotToAdd.SetFromAxisAngle(float4(0, 0, 1, 1), Time::GetDeltaTime() * rotationSpeed);
+			switchShadeTransform->SetRotation(rotToAdd * switchShadeTransform->GetGlobalRotation());
+		}
+	}
+
+	switch (switchState) {
+	case SwitchState::IDLE:
+		//This code handles the color grading progressively increasing and decreasing alpha
+		if (switchSkillParent) {
+			std::vector<GameObject*> children = switchSkillParent->GetChildren();
+			ComponentImage* fillImage = children[HIERARCHY_INDEX_SWITCH_ABILITY_FILL]->GetComponent<ComponentImage>();
+			if (fillImage) {
+
+				float delta = switchColorTimer / switchColorTotalTime;
+
+				if (switchColorIncreasing) {
+					fillImage->SetColor(float4(fillImage->GetColor().xyz(), Lerp(0.3f, 0.7f, delta)));
+				} else {
+					fillImage->SetColor(float4(fillImage->GetColor().xyz(), Lerp(0.7f, 0.3f, delta)));
+				}
+				switchColorTimer += Time::GetDeltaTime();
+			}
+		}
+
+		//Reset color timer and invert the toggle for increasing/decreasing
+		if (switchColorTimer >= switchColorTotalTime) {
+			switchColorTimer = 0;
+			switchColorIncreasing = !switchColorIncreasing;
+		}
+
+		break;
+	case SwitchState::PRE_COLLAPSE:
+
+		//TODO make glow happen
+		//TODO activate white effect under picto 
+
+		if (switchTimer > switchPreCollapseMovementTime) {
+			switchTimer = switchPreCollapseMovementTime;
+		}
+
+		if (switchGlowImage) {
+			switchGlowImage->SetColor(float4(1.0f, 1.0f, 1.0f, 1 - Clamp(switchTimer / switchPreCollapseMovementTime, 0.0f, 0.7f)));
+		}
+
+		if (fangSkillParent->IsActive()) {
+			for (int i = 0; i < skillsFang.size(); ++i) {
+				transform2D = skillsFang[i]->GetComponent<ComponentTransform2D>();
+				if (transform2D) {
+					transform2D->SetPosition(float3::Lerp(cooldownTransformOriginalPositions[i], cooldownTransformOriginalPositions[i] + float3(switchExtraOffset + i * 10.0f, 0, 0), switchTimer / switchPreCollapseMovementTime));
+				}
+
+			}
+		} else {
+			for (int i = 0; i < skillsOni.size(); ++i) {
+				transform2D = skillsOni[i]->GetComponent<ComponentTransform2D>();
+				if (transform2D) {
+					transform2D->SetPosition(float3::Lerp(cooldownTransformOriginalPositions[i + 3], cooldownTransformOriginalPositions[i + 3] + float3(switchExtraOffset + i * 10.0f, 0, 0), switchTimer / switchPreCollapseMovementTime));
+				}
+			}
+		}
+
+		if (switchTimer == switchPreCollapseMovementTime) {
+			if (switchGlowImage) {
+				switchGlowImage->SetColor(float4(1.0f, 1.0f, 1.0f, 0));
+			}
+			switchState = SwitchState::COLLAPSE;
+			switchTimer = 0;
+		}
+
+		break;
+	case SwitchState::COLLAPSE:
+		SetCooldownRetreival(Cooldowns::SWITCH_SKILL);
+
+		if (switchTimer > switchCollapseMovementTime) {
+			switchTimer = switchCollapseMovementTime;
+		}
+
+		if (switchGlowImage) {
+			switchGlowImage->SetColor(float4(1.0f, 1.0f, 1.0f, 1 - Clamp01(0.7f + switchTimer / switchPreCollapseMovementTime)));
+		}
+
+		if (fangSkillParent->IsActive()) {
+
+			if (switchTimer > switchCollapseMovementTime / 1.5f) {
```

Comment:
This and the else could be the same function switching the skillsFang to a generic target canvas just like the old swap

Reply:
Done